app/helpers/samples_helper.rb

```diff
@@ -262,26 +260,50 @@ def sample_uploader(sample)
     user
   end
 
-  def sample_derived_data(sample)
+  def sample_derived_data(sample, job_stats_hash)
     output_data = {}
     pipeline_run = sample.pipeline_runs.first
-    job_stats = pipeline_run ? pipeline_run.job_stats : nil
-    summary_stats = job_stats ? get_summary_stats(job_stats) : nil
+    summary_stats = job_stats_hash.present? ? get_summary_stats(job_stats_hash, pipeline_run) : nil
     output_data[:pipeline_run] = pipeline_run
     output_data[:host_genome_name] = sample.host_genome ? sample.host_genome.name : nil
-    output_data[:job_stats] = job_stats
     output_data[:summary_stats] = summary_stats
 
     output_data
   end
 
+  def make_job_stats_hash(pipeline_run_ids)
```

Comment:
i think this pattern is going to become more common as we run into other perf bottlenecks. at FB we used to call these functions "..._db_get" and "..._db_multiget" to make it explicit that these would trigger a db hit, and you could use the multiget version to fetch a lot at once. we might want to consider something similar here (this function would then be called job_stats_db_multiget, and there could be a job_stats_db_get function that just calls job_stats_db_multiget and returns the contained, unhashed result

Reply:
ok!